ExternalFunctions.cpp

```diff
@@ -35,6 +35,8 @@
 std::map<std::string, ExternalFunctions::RetAndArgs>
     ExternalFunctions::UserSpecifiedFunctions;
 
+std::set<std::string> ExternalFunctions::UserSpecifiedFields;
```

Comment:
This is minor, but I would suggest naming it UserSpecifiedVariables.

Reply:
That name fits better. Thanks!

Reply:
Nice. Given this addition, I think the following name changes are more representative of the data encapsulated:
1. class name `ExternalFunctions` to `IncludeFileInfo`,
2. static member `UserSpecifiedFunctions` to `ExternalFunctions`
3. static member `UserSpecifiedVariables` to `ExternalVariables`
4. static method `Function *Create(...)` to `Function *CreateFunction(...)`
5. static method `getUserSpecifiedFuncPrototypes(...)` to `getExternalFunctionPrototype(...)`
6. and finally rename the files `ExternalFunctions.*` to `IncludeFileInfo.*`

Thoughts?

Reply:
Maybe IncludedFileInfo?